jsonrpc/src/methods/history.json

```diff
@@ -0,0 +1,226 @@
+    }
+  },
+  {
+    "name": "portal_historySendPing",
+    "summary": "Send a PING message to the specified node",
+    "params": [
+      {
+        "$ref": "#/components/contentDescriptors/Enr"
+      },
+      {
+        "$ref": "#/components/contentDescriptors/DataRadius"
+      }
+    ],
+    "result": {
+      "$ref": "#/components/contentDescriptors/SendPingResult"
+    }
+  },
+  {
+    "name": "portal_historySendPong",
+    "summary": "Send a PONG message to the specified node",
+    "params": [
+      {
+        "$ref": "#/components/contentDescriptors/Enr"
+      },
+      {
+        "$ref": "#/components/contentDescriptors/RequestId"
+      },
+      {
+        "$ref": "#/components/contentDescriptors/DataRadius"
+      }
+    ],
+    "result": {
+      "$ref": "#/components/contentDescriptors/SendPongResult"
+    }
+  },
+  {
+    "name": "portal_historySendFindNodes",
+    "summary": "Send a FINDNODES request to a peer, to search within the given set of distances",
+    "params": [
+      {
+        "$ref": "#/components/contentDescriptors/Enr"
+      },
+      {
+        "$ref": "#/components/contentDescriptors/Distances"
+      }
+    ],
+    "result": {
+      "$ref": "#/components/contentDescriptors/SendFindNodeResult"
+    }
+  },
+  {
+    "name": "portal_historySendNodes",
+    "summary": "Respond to a specific FINDNODES request with a NODES response.",
+    "params": [
+      {
+        "$ref": "#/components/contentDescriptors/Enr"
+      },
+      {
+        "$ref": "#/components/contentDescriptors/Nodes"
+      },
+      {
+        "$ref": "#/components/contentDescriptors/RequestId"
+      }
+    ],
+    "result": {
+      "$ref": "#/components/contentDescriptors/SendNodesResult"
+    }
+  },
+  {
+    "name": "portal_historyPing",
+    "summary": "Send a PING message to the designated node and wait for a PONG response.",
+    "params": [
+      {
+        "$ref": "#/components/contentDescriptors/Enr"
+      },
+      {
+        "$ref": "#/components/contentDescriptors/DataRadius"
+      }
+    ],
+    "result": {
+      "$ref": "#/components/contentDescriptors/PingResult"
+    }
+  },
+  {
+    "name": "portal_historyFindNodes",
+    "summary": "Send a FINDNODES request for nodes that fall within the given set of distances, to the designated peer and wait for a response.",
+    "params": [
+      {
+        "$ref": "#/components/contentDescriptors/Enr"
+      },
+      {
+        "$ref": "#/components/contentDescriptors/Distances"
+      }
+    ],
+    "result": {
+      "$ref": "#/components/contentDescriptors/FindNodeResult"
+    }
+  },
+  {
+    "name": "portal_historyRecursiveFindNodes",
+    "summary": "Lookup a target node within in the network",
+    "params": [
+      {
+        "$ref": "#/components/contentDescriptors/NodeId"
+      }
+    ],
+    "result": {
+      "$ref": "#/components/contentDescriptors/RecursiveFindNodeResult"
+    }
+  },
+  {
+    "name": "portal_historyFindContent",
```

Comment:
@ogenev I don't see a `portal_historySendFindContent` with an additional ENR element - was there a reason for leaving that out?

Reply:
> @ogenev I don't see a `portal_historySendFindContent` with an additional ENR element - was there a reason for leaving that out?

Yeah, this is on me, I missed adding the recipient's Enr. Should be fixed now.